mp4parse/src/lib.rs

```diff
@@ -1173,129 +1173,141 @@ fn read_flac_metadata<T: Read>(src: &mut BMFFBox<T>) -> Result<FLACMetadataBlock
     })
 }
 
-fn read_esds<T: Read>(src: &mut BMFFBox<T>) -> Result<ES_Descriptor> {
+fn find_descriptor<'a>(data: &'a [u8], esds: &mut ES_Descriptor) -> Result<()> {
```

Comment:
I guess that's because the lifetime can safely propagate to the result.

I was more surprised by `let mut remains = data;` Why does that work?